samples/cpp/tutorial_code/calib3d/real_time_pose_estimation/src/main_detection.cpp

```diff
@@ -21,451 +20,483 @@
-  rmatcher.setDescriptorMatcher(matcher);                                                         // set matcher
-  rmatcher.setRatio(ratioTest); // set ratio test parameter
-
-  KalmanFilter KF;         // instantiate Kalman Filter
-  int nStates = 18;            // the number of states
-  int nMeasurements = 6;       // the number of measured states
-  int nInputs = 0;             // the number of control actions
-  double dt = 0.125;           // time between measurements (1/FPS)
-
-  initKalmanFilter(KF, nStates, nMeasurements, nInputs, dt);    // init function
-  Mat measurements(nMeasurements, 1, CV_64F); measurements.setTo(Scalar(0));
-  bool good_measurement = false;
-
-
-  // Get the MODEL INFO
-  vector<Point3f> list_points3d_model = model.get_points3d();  // list with model 3D coordinates
-  Mat descriptors_model = model.get_descriptors();                  // list with descriptors of each 3D coordinate
-
-
-  // Create & Open Window
-  namedWindow("REAL TIME DEMO", WINDOW_KEEPRATIO);
-
-
-  VideoCapture cap;                           // instantiate VideoCapture
-  cap.open(video_read_path);                      // open a recorded video
-
-  if(!cap.isOpened())   // check if we succeeded
-  {
-    cout << "Could not open the camera device" << endl;
-    return -1;
-  }
-
-  // start and end times
-  time_t start, end;
-
-  // fps calculated using number of frames / seconds
-  // floating point seconds elapsed since start
-  double fps, sec;
-
-  // frame counter
-  int counter = 0;
-
-  // start the clock
-  time(&start);
-
-  Mat frame, frame_vis;
-
-  while(cap.read(frame) && (char)waitKey(30) != 27) // capture frame until ESC is pressed
-  {
-
-    frame_vis = frame.clone();    // refresh visualisation frame
-
-
-    // -- Step 1: Robust matching between model descriptors and scene descriptors
-
-    vector<DMatch> good_matches;       // to obtain the 3D points of the model
-    vector<KeyPoint> keypoints_scene;  // to obtain the 2D points of the scene
-
-
-    if(fast_match)
+    help();
+
+    const String keys =
+            "{help h            |      | print this message                                                 }"
+            "{video v           |      | path to recorded video                                             }"
+            "{model             |      | path to yml model                                                  }"
+            "{mesh              |      | path to ply mesh                                                   }"
+            "{keypoints k       |2000  | number of keypoints to detect                                      }"
+            "{ratio r           |0.7   | threshold for ratio test                                           }"
+            "{iterations it     |500   | RANSAC maximum iterations count                                    }"
+            "{error e           |6.0   | RANSAC reprojection error                                          }"
```

Comment:
Increase RANSAC reprojection (2 pixel error seems too harsh, some valid points could be treated as outliers, especially with the incertitude errors about the intrinsic camera parameters and the camera distortion) threshold and RANSAC confidence.